<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PBR_RendererClass</class>
 <widget class="QMainWindow" name="PBR_RendererClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>PBR教学渲染器</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_6">
    <item>
     <widget class="Render" name="render"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1200</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>菜单</string>
    </property>
    <widget class="QMenu" name="menu_2">
     <property name="title">
      <string>新建默认物体</string>
     </property>
     <addaction name="actionsphere"/>
     <addaction name="actioncube"/>
    </widget>
    <addaction name="actionNewScene"/>
    <addaction name="actionOpenScene"/>
    <addaction name="actionSaveScene"/>
    <addaction name="separator"/>
    <addaction name="actionLoadModel"/>
    <addaction name="menu_2"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="functionMenu">
    <property name="title">
     <string>功能解锁</string>
    </property>
    <addaction name="actionPBRMaterial"/>
    <addaction name="separator"/>
    <addaction name="actionindirectDiffuse"/>
    <addaction name="separator"/>
    <addaction name="actionIBL"/>
   </widget>
   <widget class="QMenu" name="learnMenu">
    <property name="title">
     <string>学习</string>
    </property>
    <addaction name="actionPartone"/>
    <addaction name="actionParttwo"/>
    <addaction name="actionPartthree"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="functionMenu"/>
   <addaction name="learnMenu"/>
  </widget>
  <widget class="QDockWidget" name="sceneDockWidget">
   <property name="minimumSize">
    <size>
     <width>278</width>
     <height>300</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>350</width>
     <height>524287</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>场景与天空盒</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="scenedockWidgetContent">
    <property name="layoutDirection">
     <enum>Qt::LeftToRight</enum>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QTabWidget" name="tabWidget_2">
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="scene">
        <attribute name="title">
         <string>场景</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QTreeView" name="treeView"/>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="skybox">
        <attribute name="title">
         <string>环境贴图与光源</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QWidget" name="widget" native="true">
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="topMargin">
             <number>0</number>
            </property>
            <property name="rightMargin">
             <number>0</number>
            </property>
            <property name="bottomMargin">
             <number>0</number>
            </property>
            <item>
             <widget class="QScrollArea" name="scrollArea">
              <property name="widgetResizable">
               <bool>true</bool>
              </property>
              <widget class="QWidget" name="scrollAreaWidgetContents">
               <property name="geometry">
                <rect>
                 <x>0</x>
                 <y>0</y>
                 <width>308</width>
                 <height>322</height>
                </rect>
               </property>
               <layout class="QVBoxLayout" name="verticalLayout_8">
                <item>
                 <widget class="QWidget" name="widget_2" native="true">
                  <layout class="QVBoxLayout" name="verticalLayout_7">
                   <item>
                    <widget class="QRadioButton" name="radioButton_pointlight1">
                     <property name="enabled">
                      <bool>true</bool>
                     </property>
                     <property name="text">
                      <string>点光源1</string>
                     </property>
                     <property name="checked">
                      <bool>true</bool>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QWidget" name="widget_6" native="true">
                     <layout class="QHBoxLayout" name="horizontalLayout">
                      <item>
                       <widget class="QLabel" name="label">
                        <property name="text">
                         <string>position</string>
                        </property>
                       </widget>
                      </item>
                      <item>
                       <widget class="QLineEdit" name="lineEdit_p1px"/>
                      </item>
                      <item>
                       <widget class="QLineEdit" name="lineEdit_p1py"/>
                      </item>
                      <item>
                       <widget class="QLineEdit" name="lineEdit_p1pz"/>
                      </item>
                     </layout>
                    </widget>
                   </item>
                   <item>
                    <widget class="QWidget" name="widget_3" native="true">
                     <layout class="QHBoxLayout" name="horizontalLayout_2">
                      <item>
                       <widget class="QLabel" name="label_2">
                        <property name="text">
                         <string>color</string>
                        </property>
                       </widget>
                      </item>
                      <item>
                       <widget class="QLineEdit" name="lineEdit_p1cr"/>
                      </item>
                      <item>
                       <widget class="QLineEdit" name="lineEdit_p1cg"/>
                      </item>
                      <item>
                       <widget class="QLineEdit" name="lineEdit_p1cb"/>
                      </item>
                     </layout>
                    </widget>
                   </item>
                  </layout>
                 </widget>
                </item>
                <item>
                 <widget class="QWidget" name="widget_4" native="true">
                  <layout class="QVBoxLayout" name="verticalLayout_9">
                   <item>
                    <widget class="QRadioButton" name="radioButton_pointlight2">
                     <property name="text">
                      <string>点光源2</string>
                     </property>
                     <property name="checked">
                      <bool>true</bool>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QWidget" name="widget_7" native="true">
                     <layout class="QHBoxLayout" name="horizontalLayout_3">
                      <item>
                       <widget class="QLabel" name="label_3">
                        <property name="text">
                         <string>position</string>
                        </property>
                       </widget>
                      </item>
                      <item>
                       <widget class="QLineEdit" name="lineEdit_p2px"/>
                      </item>
                      <item>
                       <widget class="QLineEdit" name="lineEdit_p2py"/>
                      </item>
                      <item>
                       <widget class="QLineEdit" name="lineEdit_p2pz"/>
                      </item>
                     </layout>
                    </widget>
                   </item>
                   <item>
                    <widget class="QWidget" name="widget_5" native="true">
                     <layout class="QHBoxLayout" name="horizontalLayout_4">
                      <item>
                       <widget class="QLabel" name="label_4">
                        <property name="text">
                         <string>color</string>
                        </property>
                       </widget>
                      </item>
                      <item>
                       <widget class="QLineEdit" name="lineEdit_p2cr"/>
                      </item>
                      <item>
                       <widget class="QLineEdit" name="lineEdit_p2cg"/>
                      </item>
                      <item>
                       <widget class="QLineEdit" name="lineEdit_p2cb"/>
                      </item>
                     </layout>
                    </widget>
                   </item>
                  </layout>
                 </widget>
                </item>
               </layout>
              </widget>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="envButton">
              <property name="text">
               <string>选择环境贴图</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="informationDockWidget">
   <property name="minimumSize">
    <size>
     <width>200</width>
     <height>280</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>350</width>
     <height>524287</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>属性</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="informationDockWidgetContent">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="Property" name="property" native="true"/>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionNewScene">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>新建场景</string>
   </property>
  </action>
  <action name="actionOpenScene">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>打开场景</string>
   </property>
  </action>
  <action name="actionSaveScene">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>保存场景</string>
   </property>
  </action>
  <action name="actionLoadModel">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>加载模型</string>
   </property>
  </action>
  <action name="actionNewTexture">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>新建材质</string>
   </property>
  </action>
  <action name="actionLoadTexture">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>加载材质</string>
   </property>
  </action>
  <action name="actionPBRMaterial">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>PBR材质与直接光照</string>
   </property>
  </action>
  <action name="actionindirectDiffuse">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>间接漫反射</string>
   </property>
  </action>
  <action name="actionIBL">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>IBL间接镜面反射</string>
   </property>
  </action>
  <action name="actionPartone">
   <property name="text">
    <string>PBR材质部分</string>
   </property>
  </action>
  <action name="actionsphere">
   <property name="text">
    <string>球</string>
   </property>
  </action>
  <action name="actioncube">
   <property name="text">
    <string>立方体</string>
   </property>
  </action>
  <action name="actionParttwo">
   <property name="text">
    <string>间接漫反射部分</string>
   </property>
  </action>
  <action name="actionPartthree">
   <property name="text">
    <string>IBL间接镜面反射</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>Render</class>
   <extends>QOpenGLWidget</extends>
   <header>render.h</header>
  </customwidget>
  <customwidget>
   <class>Property</class>
   <extends>QWidget</extends>
   <header>property.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="PBR_Renderer.qrc"/>
 </resources>
 <connections/>
</ui>
